src/ast.ts

```diff
@@ -76,10 +70,9 @@ export class AST {
     // When an AST is to be used by CMB, it must be annotated.
     // This step is computationally intensive, and in certain instances
     // unecessary
-    if (annotate) this.annotateNodes();
-
-    this.id = -1; // just for the sake of having an id, though unused
-    this.hash = hashObject(this.rootNodes.map((node) => node.hash));
```

Comment:
Don't we need the hash, to do a quick comparison between two ASTs?

Reply:
Ooooh, good point! I suppose there's a missing optimization to be had in [the `unify` function](https://github.com/bootstrapworld/codemirror-blocks/blob/master/src/edits/patchAst.ts#L31), where if the hashes match it's a no-op.

Reply:
the unify function is already using the hashes from the ast nodes to do that it looks like, but it doesn't do it for the top of the tree (the AST instance). The missing optimization would only optimize the very first level of nodes in the tree, which probably doesn't account for much unless you have *a lot* of top level nodes.